src/lib/services/proposal/api.ts

```diff
@@ -0,0 +1,198 @@
+import axios from "axios";
+
+import {
+  zProposalAnswerCountsResponse,
+  zProposalDataResponse,
+  zProposalParamsResponse,
+  zProposalsResponse,
+  zProposalVotesInfoResponse,
+  zProposalVotesResponse,
+  zRelatedProposalsResponse,
+} from "lib/services/types";
+import type {
+  DepositParams,
+  DepositParamsInternal,
+  ProposalAnswerCountsResponse,
+  ProposalDataResponse,
+  ProposalsResponse,
+  ProposalValidatorVotesResponse,
+  ProposalVotesResponse,
+  RelatedProposalsResponse,
+  UploadAccess,
+  VotingParams,
+  VotingParamsInternal,
+} from "lib/services/types";
+import { zProposalType } from "lib/types";
+import type {
+  BechAddr,
+  BechAddr20,
+  BechAddr32,
+  Coin,
+  Option,
+  ProposalParams,
+  ProposalStatus,
+  ProposalType,
+  ProposalValidatorVote,
+  ProposalVotesInfo,
+  ProposalVoteType,
+} from "lib/types";
+import { parseWithError, snakeToCamel } from "lib/utils";
+
+export const fetchGovDepositParams = (
+  lcdEndpoint: string
+): Promise<DepositParamsInternal> =>
+  axios
+    .get<{
+      deposit_params: DepositParams;
+    }>(`${lcdEndpoint}/cosmos/gov/v1beta1/params/deposit`)
+    .then(({ data }) => snakeToCamel(data.deposit_params));
+
+export const fetchGovVotingParams = (
+  lcdEndpoint: string
+): Promise<VotingParamsInternal> =>
+  axios
+    .get<{
+      voting_params: VotingParams;
+    }>(`${lcdEndpoint}/cosmos/gov/v1beta1/params/voting`)
+    .then(({ data }) => snakeToCamel(data.voting_params));
```

Comment:
Let's apply Zod

Reply:
Mostly of these 2 functions are still using interface. I think we should separate new PR in next sprint.